gtsfm/common/sfm_result.py

```diff
@@ -26,25 +28,22 @@ def __eq__(self, other: object) -> bool:
         if not isinstance(other, SfmResult):
             return False
 
-        if not self.sfm_data.equals(other.sfm_data, 1e-9):
+        if self.gtsfm_data != other.gtsfm_data:
             return False
 
         # finally, compare reprojection error
-        return np.isclose(
-            self.total_reproj_error,
-            other.total_reproj_error,
-            rtol=1e-2,
-            atol=1e-1,
-            equal_nan=True,
-        )
+        return np.isclose(self.total_reproj_error, other.total_reproj_error, rtol=1e-2, atol=1e-1, equal_nan=True)
 
     def get_camera_poses(self) -> List[Pose3]:
         """Getter for camera poses wTi.
 
         Returns:
             camera poses as a list, each representing wTi
         """
-        return [self.sfm_data.camera(i).pose() for i in range(self.sfm_data.number_cameras())]
+        cameras = [self.gtsfm_data.get_camera(i) for i in range(self.gtsfm_data.number_images())]
```

Comment:
why not do:
```python
cameras = [ self.gtsfm_data.get_camera(i) for i in self.gtsfm_data.get_valid_camera_indices() ]
```
Isn't that simpler?

Reply:
That won't work because we need None for a camera if it wasn't optimized.

Reply:
I see. Then can we make 2 changes:

The return signatures should be `List[Optional[Pose3]]`
Can we say that the length here should be equal to the number of images in `gtsfm_data`, even if they are None